great_expectations/datasource/fluent/interfaces.py

```diff
@@ -516,6 +516,9 @@ def delete_asset(self, asset_name: str) -> None:
         asset: _DataAssetT
         self.assets = list(filter(lambda asset: asset.name != asset_name, self.assets))
 
+        if self._data_context:
```

Comment:
Maybe we should make a private `_save_context` method that does this conditional check for us.